src/test/java/org/prebid/server/floors/BasicPriceFloorAdjusterTest.java

```diff
@@ -0,0 +1,129 @@
+package org.prebid.server.floors;
+
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.prebid.server.VertxTest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestBidadjustmentfactors;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+import org.prebid.server.proto.openrtb.ext.request.ImpMediaType;
+
+import java.math.BigDecimal;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+
+import static java.util.function.UnaryOperator.identity;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BasicPriceFloorAdjusterTest extends VertxTest {
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    private BasicPriceFloorAdjuster basicPriceFloorAdjuster;
+
+    @Before
+    public void setUp() {
+        basicPriceFloorAdjuster = new BasicPriceFloorAdjuster();
+    }
+
+    @Test
+    public void adjustForImpShouldApplyFactorToBidFloorIfPresent() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(identity());
+        final Imp imp = givenImp(identity());
+        final String bidder = "bidder";
```

Comment:
According to project style, just create 
```java
private static final String RUBICON = "rubicon";
```
And then use it directly in when block, also actually everything can be moved to `when` block